optimum/commands/base.py

```diff
@@ -0,0 +1,144 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Optimum command-line interface base classes."""
+
+from abc import ABC
+from argparse import ArgumentParser, RawTextHelpFormatter
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Optional, Tuple, Type
+
+
+if TYPE_CHECKING:
+    from argparse import Namespace, _SubParsersAction
+
+
+@dataclass(frozen=True)
+class CommandInfo:
+    name: str
+    help: str
+    subcommand_class: Optional[Type["BaseOptimumCLICommand"]] = None
+    formatter_class: Type = RawTextHelpFormatter
+
+    @property
+    def is_subcommand_info(self):
+        return self.subcommand_class is not None
+
+    def is_subcommand_info_or_raise(self):
+        if not self.is_subcommand_info:
+            raise ValueError(f"The command info must define a subcommand_class attribute, but got: {self}.")
+
+
+class InvalidCLICommand(Exception):
```

Comment:
When and how this class is used?

Reply:
It was used before but not anymore, removed it.